hera_qm/utils.py

```diff
@@ -635,9 +687,42 @@ def or_collapse(a, weights=None, axis=None, returned=False):
         return o
 
 
+def and_collapse(a, weights=None, axis=None, returned=False):
+    ''' Function to collapse axes using AND operation
+    Args:
+        a - boolean array to process
+        weights - NOT USED, but kept for symmetry with other averaging functions
+        axis - axis or axes over which to OR
+        returned - whether to return dummy weights array. NOTE: the dummy weights
+                   will simply be an array of ones. Default is False.
+    '''
+    if a.dtype != np.bool:
+        raise ValueError('Input to and_collapse function must be boolean array')
+    o = np.all(a, axis=axis)
+    if (weights is not None) and not np.all(weights == weights.reshape(-1)[0]):
+        warnings.warn('Currently weights are not handled when AND-ing boolean arrays.')
+    if returned:
+        return o, np.ones_like(o, dtype=np.float)
+    else:
+        return o
+
+
+def and_rows_cols(waterfall):
+    """ For a 2D flag waterfall, flag pixels only if fully flagged along
+    time and/or frequency
+    Args:
+        waterfall - 2D boolean array of shape (Ntimes, Nfreqs)
+    """
+    wf = np.zeros_like(waterfall, dtype=np.bool)
+    Ntimes, Nfreqs = waterfall.shape
+    wf[:, (np.sum(waterfall, axis=0) / Ntimes) == 1] = True
+    wf[(np.sum(waterfall, axis=1) / Nfreqs) == 1] = True
+    return wf
+
+
 # Dictionary to map different methods for averaging data.
 averaging_dict = {'mean': mean, 'absmean': absmean, 'quadmean': quadmean,
```

Comment:
Structurally, I find this a bit weird. You've gone out of your way to make these all have the same input format. I think you'd save a lot of complexity if you just combined them all into one function with a method kwarg.